Easy admin bundle

pueppiblue opened the pull request.

pueppiblue committed:
composer update #7

  - Removing doctrine/doctrine-bundle (1.6.3)
  - Installing doctrine/doctrine-bundle (1.6.4)
  - Removing webmozart/assert (1.0.2)
  - Installing webmozart/assert (1.1.0)

pueppiblue committed:
install EasyAdminBundle and add basic configuration #7

  install dependencies:
    Installing twig/extensions (v1.3.0)
    Installing pagerfanta/pagerfanta (v1.0.3)
    Installing javiereguiluz/easyadmin-bundle (v1.15.1)
  activate EasyAdminBundle in AppKernel
  add eaysadmin routes to routing.yml
  add easyadmin config file and import it in config.yml
  give ROLE_ADMIN access to /admin route

pueppiblue committed:
change tablename to clm_character #7

  change tablename in orm schema config ClmCharacter.orm.yml
  adjust tablename in migration
  set password for clm_test db in config_test.yml

pueppiblue committed:
activate doctrine cache for prod environment #7

pueppiblue committed:
integrate easyadmin with FOSUserBundle #7

  add custom config for User form and list views
  change menu design
  override AdminController and methods for User entity
  change eadmin route to /eadmin

pueppiblue committed:
adjust entities for eadmin forms #7

  Make name parameter optiononal for entity constructor.
  Add setters for name property.
  Add __toString() method to entity for data transformation.
  Fix class name of DoctrineORMClmItemRepository.
  Change eadmin config to show password in form for User entity.

pueppiblue committed:
let eadmin manage ClmItem entity #7

  add ClmItem entity to eadmin config.
  add Items as menu entry.
  make name parameter in ClmItem constructor optional.
  change default value of isCash property to false.

pueppiblue committed:
add minus-circle icon to delete action #7

pueppiblue committed:
update shippable mysql privilege settings

pueppiblue committed:
fix mysql syntax in shippable.yml

robertfausk commented on app/DoctrineMigrations/Version20160803183748.php:
```diff
@@ -18,11 +18,11 @@ public function up(Schema $schema)
         $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
 
-        $this->addSql('CREATE TABLE clm_characters (id INT AUTO_INCREMENT NOT NULL, user_id INT DEFAULT NULL, character_name VARCHAR(190) NOT NULL, clm_class VARCHAR(190) NOT NULL, preferred_set VARCHAR(190) DEFAULT NULL, INDEX IDX_3156DDECA76ED395 (user_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE clm_character (id INT AUTO_INCREMENT NOT NULL, user_id INT DEFAULT NULL, character_name VARCHAR(190) NOT NULL, clm_class VARCHAR(190) NOT NULL, preferred_set VARCHAR(190) DEFAULT NULL, INDEX IDX_3156DDECA76ED395 (user_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = InnoDB');
```
Create new migrations instead of editing old ones.

pueppiblue replied:
Fox complex changes more fail safe. But in this case the ROI was not good enough for me to put up an entire migration for a simple character difference.

robertfausk commented on app/config/config.yml:
```diff
@@ -2,6 +2,7 @@ imports:
     - { resource: parameters.yml }
     - { resource: security.yml }
     - { resource: services.yml }
+    - { resource: config_eadmin.yml }
```
Wording ```config_eadmin.yml``` confuses cause it's similar to ```config_dev.yml```.
I think ```eadmin.yml``` would be better.

robertfausk commented on src/AppBundle/Entity/ClmAccount.php:
```diff
@@ -50,14 +50,32 @@ class ClmAccount
      * @param string $name
      */
-    public function __construct($name)
+    public function __construct($name = '')
```
Should it be possible to have an empty name?

pueppiblue replied:
EAdmin instantiates the object upon rendering the "new" form and thus cannot provide a name yet.
I could override the createNewClmAccountEntity() , but i would have to provide a dummy name there which then will be overwritten by the form input. Not very elegant either.
On the other hand $name is part of the primary key and thus required to persist the object to the database. I think its just the way easyadmin handles the entities.
Both ways will work. I'm eager for other suggestions from your side.

robertfausk replied:
The attribute ```accountName``` isn't part of id (and also shouldn't be) but it's not nullable so it's true that it have to be set to persist the object into database.

I would prefer to omit setting params in constructor as long as we do not use DDD approach or have another good reason to do so.

robertfausk commented on src/AppBundle/Entity/ClmCharacter.php:
```diff
@@ -38,13 +38,21 @@ class ClmCharacter
      * @param $clmClass
      */
-    public function __construct($name, $clmClass = '')
+    public function __construct($name = '', $clmClass = '')
```
Should it be possible to have an empty name and an empty clmClass?

robertfausk commented on src/AppBundle/Entity/ClmItem.php:
```diff
@@ -35,17 +35,24 @@ class ClmItem
      * @param string $name
      */
-    public function __construct($name)
+    public function __construct($name = '')
```
Should it be possible to have an empty name?

pueppiblue committed:
rename config name for eadmin bundle #49

pueppiblue committed:
rewrite/remove entity constructors #47

  Since character.class is not set anymore by the constructor
  the schema is changed to let class be nullable.

pueppiblue committed:
move var declaration to top and add function annotation #47

pueppiblue committed:
change mysql version to 5.6 #49

  shippable uses mysql 5.6 which does not support
  ALTER TABLE ... RENAME INDEX...
  Remove 5.7 migration and add new migration.

pueppiblue committed:
change faker method for unique account name

  Shippable creates an  duplicate entry integrity constraint violation
  despite setting (unique) flag on accountName property in clmAccounts.yml
    [Doctrine\DBAL\Exception\UniqueConstraintViolationException]
    Integrity constraint violation: 1062
    Duplicate entry 'Änne.test' for key 'UNIQ_A81AAB685E237E06'

pueppiblue committed:
Merge branch 'master' into easy_admin_bundle

# Conflicts:
#	shippable.yml
#	src/AppBundle/DataFixtures/Faker/Provider/ConanItemProvider.php

pueppiblue committed:
create new single migration from blank database #49
